core/src/main/java/org/eclipse/hono/config/HonoConfigProperties.java

```diff
@@ -322,4 +322,23 @@ public HonoConfigProperties setWaitForDownstreamConnectionEnabled(final boolean
         this.waitForDownstreamConnection = waitForConnection;
         return this;
     }
+
+    public boolean handleUnconfiguredPort() {
+        boolean configuredPort = false;
+        if (getPort() == HonoConfigProperties.PORT_UNCONFIGURED) {
+            setPort(Constants.PORT_AMQP);
```

Comment:
Since we want to use this class for configuring protocol adapters as well, I do not think that it makes much sense to *hardwire* the default AMQP port here. Instead, we probably should have a method like `getPort(int defaultPort)` which sets the port to the passed in default value if it is unconfigured ...